invalid memory address or nil pointer dereference

2022/02/28 08:41:19 Starting recording
2022/02/28 08:41:19 File already exists
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x620c9e]

goroutine 36 [running]:
github.com/jawee/twitch-recorder/internal/postprocessor.(*FileMovePostProcessor).Process(0xc000010078, 0x0)
        /usr/src/app/internal/postprocessor/postprocessor.go:29 +0x3e
created by main.startPostProcessing
        /usr/src/app/cmd/main.go:56 +0x45
2022/02/28 08:42:19 New. Loading configuration

--- a/internal/processor/processor.go
+++ b/internal/processor/processor.go
@@ -65,8 +65,9 @@ func (sp *StreamProcessor) ProcessStreamer(username string) error {
                 res, err := sp.rec.Record(user.DisplayName, filename)
                 if err != nil {
                     log.Println(err)
+                } else {
+                    sp.c <- res
                 }
-                sp.c <- res
                 // Do I want a channel for each download, or just one?
                 //close(sp.c)
             }()

--- a/internal/recorder/recorder.go
+++ b/internal/recorder/recorder.go
@@ -1,6 +1,7 @@
 package recorder
 
 import (
+	"errors"
 	"log"
 	"os"
 	"os/exec"
@@ -38,8 +39,7 @@ func (s* StreamlinkRecorder) Record(username string, filename string) (*Recorded
 
     _, err := os.Stat(filePath)
     if err == nil {
-        log.Println("File already exists")
-        return nil, err
+        return nil, errors.New("File already exists")
     }
 
     s.notificationClient.SendMessage("Starting recording for " + username + ". File " + filename)

